File: app/src/main/java/org/codroid/body/widget/Divider.java
package org.codroid.body.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.RectF;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.Nullable;

import org.codroid.body.R;

public class Divider extends View {

    private Paint mPaint;
    private RectF rectF;
    private int color;

    public Divider(Context context) {
        super(context);
    }

    public Divider(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public Divider(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    public Divider(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init(context, attrs);
    }

    private void init(Context context, @Nullable AttributeSet attrs) {
        TypedArray array = context.obtainStyledAttributes(attrs, R.styleable.Divider);
        color = array.getColor(R.styleable.Divider_backgroundColor, Color.BLACK);
        mPaint = new Paint();
        mPaint.setStyle(Paint.Style.FILL);
        mPaint.setColor(color);
        array.recycle();
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (rectF == null) rectF = new RectF(0, 0, getWidth(), getHeight());

        canvas.drawRoundRect(rectF, 10, 10, mPaint);
    }
}

File: app/src/main/kotlin/org/codroid/body/widgets/Divider.kt
package org.codroid.body.widgets

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.graphics.RectF
import android.util.AttributeSet
import android.view.View
import org.codroid.body.R

class Divider : View {
    constructor(context: Context?) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initialize(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initialize(context, attrs)
    }

    private val mPaint = Paint()
    private val mRect = RectF()

    private var mLineSize = 0

    private fun initialize(context: Context, attrs: AttributeSet?) {
        val ta = context.obtainStyledAttributes(attrs, R.styleable.Divider)
        mPaint.color = ta.getColor(R.styleable.Divider_backgroundColor, Color.BLACK)
        mPaint.style = Paint.Style.FILL
        mPaint.isAntiAlias = true
        mLineSize = ta.getDimensionPixelSize(R.styleable.Divider_lineSize, 10)
        ta.recycle()
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val widthMode = MeasureSpec.getMode(widthMeasureSpec)
        val widthSize = MeasureSpec.getSize(widthMeasureSpec)
        val heightMode = MeasureSpec.getMode(heightMeasureSpec)
        val heightSize = MeasureSpec.getSize(heightMeasureSpec)

        var width = mLineSize + paddingLeft + paddingRight
        var height = mLineSize + paddingTop + paddingBottom
        if (widthMode == MeasureSpec.EXACTLY) {
            width = widthSize
        } else if (widthMode == MeasureSpec.AT_MOST) {
            width = widthSize.coerceAtMost(width)
        }

        if (heightMode == MeasureSpec.EXACTLY) {
            height = heightSize
        } else if (heightMode == MeasureSpec.AT_MOST) {
            height = heightSize.coerceAtMost(height)
        }
        mRect.left = paddingLeft.toFloat()
        mRect.top = paddingTop.toFloat()
        mRect.right = width - paddingRight.toFloat()
        mRect.bottom = height - paddingBottom.toFloat()
        setMeasuredDimension(width, height)
    }

    override fun onDraw(canvas: Canvas?) {
        canvas?.run {
            drawRoundRect(mRect, 10f, 10f, mPaint)
        }
    }
}